### ISSUE TITLE
False Positive `reportNoOverloadImplementation` after passing function through generics

### ISSUE THREAD

**Issue by Alan-Chen99** [microsoft/pyright#9803]
```python
from typing import Generic, TypeVar

T = TypeVar("T")


class cast(Generic[T]):
    def __init__(self, _: T):
        pass

    def __call__(self, a) -> T:
        return a


@cast(print)
def print(  # error: "print" is marked as overload, but no implementation is provided (reportNoOverloadImplementation)
    *args, **kwargs
):
    assert False


# workaround
def wrap(x: T) -> T:
    assert False


print = wrap(print)
```

```
pyright --version
pyright 1.1.393
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.394.
